src/webcam/react-webcam.tsx

```diff
@@ -0,0 +1,473 @@
+    if (!shouldUseBackCam) return Promise.resolve(constraints)
+    return enumerateDevices().then((devices: DeviceData[]) => {
+      const cameras = extractCamerasFromDevices(devices)
+
+      const mainBackCam = mainBackCamera(cameras)
+      if (mainBackCam && mainBackCam.deviceId === '') {
+        if (!constraints.video) {
+          constraints.video = {}
+        }
+        // @ts-ignore
+        constraints.video.facingMode = { ideal: 'environment' }
+      } else {
+        const deviceId =
+          mainBackCam && mainBackCam.deviceId
+            ? mainBackCam.deviceId
+            : cameras[0].deviceId
+        // @ts-ignore
+        constraints.video.deviceId = { exact: deviceId }
+      }
+      return constraints
+    })
+  }
+}
+
+const isBackCameraLabel = (label: string): boolean => {
+  const lowercaseLabel = label.toLowerCase()
+
+  return backCameraKeywords.some((keyword) => {
+    return lowercaseLabel.includes(keyword)
+  })
+}
+
+const cameraObjects = new Map()
+
+type CameraType = {
+  deviceId?: string
+  label: string
+  cameraType: 'front' | 'back'
+}
+
+const extractCamerasFromDevices = (devices: DeviceData[]): CameraType[] => {
+  const cameras: Array<CameraType> = devices
+    .filter((device: DeviceData) => {
+      return device.kind === 'videoinput'
+    })
+    .map((videoDevice: DeviceData) => {
+      if (cameraObjects.has(videoDevice.deviceId)) {
+        return cameraObjects.get(videoDevice.deviceId)
+      }
+
+      const label = videoDevice.label || ''
+      const camera = {
+        deviceId: videoDevice.deviceId,
+        label,
+        cameraType: isBackCameraLabel(label) ? 'back' : 'front',
+      }
+
+      if (label !== '') {
+        cameraObjects.set(videoDevice.deviceId, camera)
+      }
+
+      return camera
+    })
+  if (
+    cameras.length > 1 &&
+    !cameras.some((camera) => {
+      return camera.cameraType === 'back'
+    })
+  ) {
+    // Check if cameras are labeled with resolution information, take the higher-resolution one in that case
+    // Otherwise pick the last camera
+    let backCameraIndex = cameras.length - 1
+
+    const cameraResolutions = cameras.map((camera) => {
+      const match = camera.label.match(/\b([0-9]+)MP?\b/i)
+      if (match) {
+        return parseInt(match[1], 10)
+      }
+
+      return NaN
+    })
+    if (
+      !cameraResolutions.some((cameraResolution) => {
+        return isNaN(cameraResolution)
+      })
+    ) {
+      backCameraIndex = cameraResolutions.lastIndexOf(
+        Math.max(...cameraResolutions)
+      )
+    }
+    cameras[backCameraIndex].cameraType = 'back'
+  }
+
+  return cameras
+}
+
+const mainBackCamera = (cameras: Array<CameraType>) =>
+  cameras
+    .filter((camera) => camera.cameraType === 'back')
+    .sort((camera1, camera2) => camera1.label.localeCompare(camera2.label))[0]
+
+const DEBUG = false
+const debugConsole = (...args: any[]) => {
+  if (DEBUG) console.log(...args)
```

Comment:
Instead of a var, we can use `process.env.NODE_ENV=== 'development'` to log in development by default?

Reply:
done